From: Petter Reinholdtsen <pere@debian.org>
Subject: Avoid crash in lcPiecesLibrary::OpenCache() when opening project.
  Make sure all class variables are initialized, and do not call stat()
  when no  cache file name is set.
Origin: vendor
Forwarded: no

---
 common/lc_library.cpp |    7 +++++++
 1 file changed, 7 insertions(+)

--- leocad.orig/common/lc_library.cpp
+++ leocad/common/lc_library.cpp
@@ -18,6 +18,13 @@
 lcPiecesLibrary::lcPiecesLibrary()
 {
 	mNumOfficialPieces = 0;
+	mLibraryPath[0] = 0;
+	mCacheFileName[0] = 0;
+	mCategoriesFile[0] = 0;
+	mCacheFileModifiedTime = 0;
+	mLibraryFileName[0] = 0;
+	mCategoriesModified = false;
+
 	mZipFile = NULL;
 	mCacheFile = NULL;
 	mCacheFileName[0] = 0;
